build-system/babel-plugins/babel-plugin-transform-jss/index.js

```diff
@@ -126,3 +127,18 @@ module.exports = function ({types: t, template}) {
     },
   };
 };
+
+// Abuses spawnSync to let us run an async function sync.
+function transformCssSync(cssText) {
+  const programText = `
```

Comment:
Can we actually do it like this? Or should we be calling a gulp task? I.e. what if the `jsify-css` starts using some gulp API?

Reply:
I'm not sure what you mean. If `jsify-css` changes in a way that breaks this function, the tests will fail.

I don't know how we could pass all of this as an arg to a gulp task and whether or not we can get the response back in an sensible way. Justin alerted me that we have some exec helpers in build system I'll check out.

Reply:
Basically my question is only: our build system is built on gulp and I'm not sure how valid it is to run any part of it as an independent executable. If it works, it's probably ok, but I'd definitely check with the infra team.

Reply:
> Justin alerted me that we have some exec helpers in build system I'll check out.

Ah, can't use these just yet. Need to update the exec helpers to support the args concept, since each platform needs different args escaping.

Reply:
When we spawn a process like this, don't we get access to the stdin/stdout file handles? My hunch is maybe this could be a standalone script that reads from stdin and transforms that CSS. That in turn could play nicer with gulp/parallelization, and completely avoid the escaping-command-line-arguments issue. WDYT(?

Reply:
It could be! I'm not sure if that helps though, since it would only cause more latency for the gulp css task, and this already solves the encoding issue for the babel transform.

Reply:
Piling on to this thread with some alternative ideas. First off, do I understand correctly that if `transformCss()` from `jsify-css.js` were synchronous, we wouldn't need the workaround in this file? And is it true that the thing that makes `transformCss()` asynchronous is this call to `postcss().process()`?

https://github.com/ampproject/amphtml/blob/32579ce377e0849f04966951fc3123aef8f751d9/build-system/tasks/jsify-css.js#L75-L77

Here are two alternatives that come to mind:
1. From looking at this code, can we use `postcss-js`, which appears to have a [synchronous API](https://github.com/postcss/postcss-js#syncplugins-function)?
2. Modify the code in `jsify-css.js` to work [directly on the `gulp` stream](https://www.npmjs.com/package/postcss#gulp), and wrap it to create a synchronous function that can be called here.

Reply:
Yep, if it were sync then we wouldn't need to call `spawnSync` here, and `postcss` is what causes async.

1. I tried the sync api, but it throws since all of our plugins can only operate in async mode :/
2. I don't know enough about `gulp` to say if this makes sense, but if the streams abstraction lets us easily turn this async function sync (via abstractions over subprocesses), then that would definitely be better! Only better for the babel transforms though, the previous callers were fine async